plugins/link/plugin.js

```diff
@@ -94,10 +94,11 @@
 			editor.on( 'doubleclick', function( evt ) {
 				// If the link has descendants and the last part of it is also a part of a word partially
 				// unlinked, clicked element may be a descendant of the link, not the link itself. (http://dev.ckeditor.com/ticket/11956)
-				var element = CKEDITOR.plugins.link.getSelectedLink( editor ) || evt.data.element.getAscendant( 'a', 1 );
+				// evt.data.element condition allows opening anchor dialog if the anchor is empty (#501).
+				var element = CKEDITOR.plugins.link.getSelectedLink( editor ) || evt.data.element.getAscendant( 'a', 1 ) || evt.data.element;
```

Comment:
I wonder if this condition could be simplified, e.g. setting `element` only if `evt.data.element` is `a` or `img`.